test/utils/test_transforms.py

```diff
@@ -262,6 +262,48 @@ def test_separate_circuit(self):
                 [("A", 0), ("B", 0), ("C", 0)], separated_circuits.qubit_map
             )
 
+        with self.subTest("barriers are ignored"):
+            qreg = QuantumRegister(3)
+            circuit = QuantumCircuit(qreg)
+            circuit.h(0)
+            circuit.x(0)
+            circuit.barrier(0)
+            circuit.x(1)
+            circuit.h(1)
+            circuit.y(2)
+            circuit.h(2)
+
+            separated_circuits = separate_circuit(circuit, partition_labels="ABC")
```

Comment:
Actually, I believe this test does not actually check that the relevant bug is fixed.

The idle wire detection code only runs when `partition_labels` is not provided (or is `None`).  In this case, the partition labels are determined automatically from the connectivity of the circuit.   #594 improved on this code so that idle wires are removed from the circuit during separation, but the problem is that any qubit with a barrier (e.g., if a barrier is placed over all qubits, even the idle ones) will be falsely detected as an active qubit.  Your change should indeed fix #621 but the test does not ensure that it is fixed.  The test is actually _almost_ testing the check desired in #626, but not quite.